api/serializers.py

```diff
@@ -233,27 +235,40 @@ def get_sources(self, obj):
         return SourceSerializer(video.get('sources'), many=True).data
 
 
-class MediaAnalyticsItemSerializer(serializers.Serializer):
+class MediaItemAnalyticsSerializer(serializers.Serializer):
     """
     The number of viewing for a particular media item on a particular day.
 
     """
-    date = serializers.SerializerMethodField(help_text='The day when a media was viewed')
-    views = serializers.SerializerMethodField(help_text='The number of media views on a day')
+    date = serializers.DateField(
+        source='day', help_text='The day when a media was viewed', read_only=True)
+    views = serializers.IntegerField(
+        source='num_hits', help_text='The number of media views on a day', read_only=True)
 
-    def get_date(self, row):
-        return str(row[0])
 
-    def get_views(self, row):
-        return row[1]
-
-
-class MediaAnalyticsListSerializer(serializers.Serializer):
+class MediaItemAnalyticsListSerializer(serializers.Serializer):
     """
     A list of media analytics data points.
 
     """
-    results = MediaAnalyticsItemSerializer(many=True, source='*')
+    results = serializers.SerializerMethodField()
+
+    ResultRow = namedtuple('ResultRow', 'day num_hits')
+
+    def get_results(self, obj):
```

Comment:
Ordinarily, yes, but since there's no corresponding Django model for the analysitcs data, a query set cannot be produced. OTOH, it does make sense to try and make the query in the view. I'll have a go.